### PR TITLE
Fix currency formatting for totals on My Account > Subscriptions page

### LINKED ISSUE #5488: Subscriptions decimal points format is unexpectedly governed by customer currency
### Describe the bug

When customer currency is set to zero-decimal currency, eg. JPY, the price format on My Account > Subscriptions page is changed accordingly. This is unexpected. I think the currency format should follow the currency of when the subscription was purchased.

### To Reproduce

1. Enable [multi currencies](https://woocommerce.com/document/woocommerce-payments/currencies/multi-currency-setup/#section-2) and add some currencies, eg `EUR` and `JPY`.
2. Buy a couple of subscriptions with different currencies. You can switch currency by adding `currency` parameter in the query string of a shop or product page, eg. `/shop?currency=JPY`
3. Go to My Account > Subscriptions page.

### Actual behavior

Notice when customer currency is set to a zero-decimal currency, ie `/shop?currency=JPY`, the prices shown have no decimal points. When customer currency is set to a non zero-decimal currency, ie `/shop?currency=USD`, the prices shown have decimal points.

### Screenshots

https://user-images.githubusercontent.com/73803630/215522440-fa388e48-284e-4345-8a36-2625e102b06c.mov

### Expected behavior

The currency format should follow the currency of when the subscription was purchased.

### Additional context

This might not only affect subscription but it was what I observed when I was testing a PR (https://github.com/Automattic/woocommerce-payments/pull/5432).

### PR DESCRIPTION
**Pull request by jessepearson**
Fixes #5488

#### Changes proposed in this Pull Request

* Remove `static` declaration for `MultiCurrency->is_initialized()` method.
* Add `current_my_account_subscription` property to hold the current subscription being iterated through on the My Account > Subscriptions page.
* Add `frontend_currencies` property for the `FrontendCurrencies` class/object to be used in the class.
* Add three new filters. Two filters are used as trigger events, and then the `wc_price` filter will add the 3 letter currency code, if needed.
* Add another check in `override_selected_currency` method to check if `current_my_account_subscription` is set, and if so, use that subscription's currency.
* Add `maybe_set_current_my_account_subscription` method that is called by `woocommerce_subscription_price_string_details` filter. This will set the value of `current_my_account_subscription` to the passed subscription if the backtrace calls are found, and it does not actually filter any values.
* Add `maybe_clear_current_my_account_subscription` method that is called by `woocommerce_get_formatted_subscription_total` filter. This will set the value of `current_my_account_subscription` to `null` if the backtrace calls are found, and it does not actually filter any values.
* Add `maybe_get_explicit_format_for_subscription_total` method that checks for a subscription in `current_my_account_subscription`, and if found it will use `WC_Payments_Explicit_Price_Formatter::get_explicit_price_with_currency` to add the three letter currency code to the price string.
* Add `is_current_my_account_subscription_set` method to check to see if `current_my_account_subscription` is a subscription.

I added a note regarding a test that is currently marked as skipped due to I could not find a way to make it work.

#### Testing instructions

> [!Note]
> This requires the Woo Subscriptions plugin to be installed.
> There is a bug with rounding if a zero decimal currency is selected: https://github.com/Automattic/woocommerce-payments/issues/7537 

* Do not check out this branch yet.
* If you do not have subscriptions active with different currencies on your test site, you will need to create them.
  * If needed: Navigate to WooCommerce > Settings > Multi-Currency and add other currencies, such as: EUR, GBP, JPY.
  * If needed: Navigate to Products > Add New and add a new subscription for 100 a day.
  * If needed: Navigate to your subscription on the front end and purchase in multiple currencies.
* Navigate to My Account > Subscriptions.
* Change the currency to JPY and you will see the formatting of the totals for the currencies change to the JPY format.
* Check out this branch, then refresh the page.
* Change the currency to multiple different currencies, the format for the total amounts should remain proper for the currency itself and should not take on the formatting for the currency selected.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7408` or branch name `fix/5488-subs-page-currency-format-incorrect` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/5488-subs-page-currency-format-incorrect) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 86755403d95f188210e3552ee625c50692b7627e
- Build time: 2023-11-01 13:39:19 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/unit/multi-currency/compatibility/test-class-woocommerce-subscriptions.php [hunk 1]
```diff
@@ -760,6 +806,52 @@ public function test_should_disable_currency_switching_return_false_when_startin
...
+	 * This test is marked as skipped due to it does not currently work.
+	 *
+	 * It does not work due to WC_Payments_Explicit_Price_Formatter::should_output_explicit_price is called in the path
+	 * of maybe_get_explicit_format_for_subscription_total, and when it is called it loads the MultiCurrency instance
+	 * that is initialized before the test itself starts. The self::$multi_currency_instance->has_additional_currencies_enabled()
+	 * check returns false due to no other currencies are enabled because there's no Stripe connection.
+	 *
+	 * I could not find a way to properly mock this due to the static call to the MC instance.
+	 */
+	public function test_maybe_get_explicit_format_for_subscription_total() {
```
Comment (jessepearson) [inline tests/unit/multi-currency/compatibility/test-class-woocommerce-subscriptions.php:809]: I am not sure if we should leave this test in and have it skipped, or remove it. I could not find a way to mock it due to the reasons mentioned in the comment.

**Review by tpaksu — APPROVED** [review]
> [!NOTE] 
> Tested while USD was the active currency on the store.

### Before the fix was applied

#### My subscriptions page
image

#### Subscription details page
image

### After the fix

#### My subscriptions page
image

#### Subscription details page
image

The subsription details page shows the price with the USD format, but I guess this only applies to the Subscriptions list. Approving this as it works as it's described within the scope of it. 

The code is very well structured, readable and understandable! Thanks for working on this and fixing it. LGTM, we can ship it.

File: tests/unit/multi-currency/compatibility/test-class-woocommerce-subscriptions.php (hunk 1 above)
Comment (jessepearson) [reply tests/unit/multi-currency/compatibility/test-class-woocommerce-subscriptions.php:809]: We talked through this test and how to fix it in Slack: p1698840378803849-slack-C01B8KNUYSW

**Pull request merged**